### PR TITLE
refactor(module:*): migrate the source code to Angular v10

### PR DESCRIPTION
**Pull request by hsuanxyz**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/core/services/scroll.ts [hunk 1]
```diff
@@ -103,13 +105,3 @@ export class NzScrollService {
     reqAnimFrame(frameFunc);
   }
 }
-
-export function SCROLL_SERVICE_PROVIDER_FACTORY(doc: Document, scrollService: NzScrollService): NzScrollService {
-  return scrollService || new NzScrollService(doc);
-}
-
-export const SCROLL_SERVICE_PROVIDER: Provider = {
-  provide: NzScrollService,
```
Comment (hsuanxyz) [inline components/core/services/scroll.ts:112]: @cipchk 这里升 10 会报错。所以拿掉了直接`providedIn: 'root'` 也没发现什么问题。这块儿还有其他的影响吗？

File: components/message/base.ts [hunk 2]
```diff
@@ -59,6 +59,8 @@ export abstract class NzMNService {
   }
 }
 
+@Directive()
+// tslint:disable-next-line:directive-class-suffix
 export abstract class NzMNContainerComponent implements OnInit, OnDestroy {
```
Comment (hsuanxyz) [inline components/message/base.ts:64]: 按文档的说法基类都用空 `Directive`, 不然 `Component` 必须写 metadata，还要声明在 `declarations`

File: components/core/services/scroll.ts (hunk 1 above)
Comment (cipchk) [reply components/core/services/scroll.ts:112]: 没有问题，这样反而更简约。

File: components/message/base.ts (hunk 2 above)
Comment (cipchk) [reply components/message/base.ts:64]: 我帮忙加个链接：https://angular.io/guide/migration-undecorated-classes#what-does-this-migration-do

Comment (cipchk) [reply components/message/base.ts:64]: 我的理解是，但凡基类有包含 `@HostBinding()`、`@ViewChild()`、`@Input()`、`@Output()` 带有这类 Angular 特征都需要加上空的 `@Directive()` 或 `@Component()`。

Comment (hsuanxyz) [reply components/message/base.ts]: 嗯，我们不考虑向下兼容 `BaseModalContainer` `NzMNContainerComponent` `NzMNComponent` 可以拿掉。

**Pull request merged**